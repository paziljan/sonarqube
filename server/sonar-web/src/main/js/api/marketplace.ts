/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { checkStatus, corsRequest, getJSON, parseJSON, postJSON } from '../helpers/request';
import throwGlobalError from '../app/utils/throwGlobalError';

export interface Edition {
  key: string;
  name: string;
  desc: string;
  more_link: string;
  request_license_link: string;
  download_link: string;
}

export interface EditionsPerVersion {
  [version: string]: Edition[];
}

export interface EditionStatus {
  currentEditionKey?: string;
  nextEditionKey?: string;
  installationStatus:
    | 'NONE'
    | 'AUTOMATIC_IN_PROGRESS'
    | 'MANUAL_IN_PROGRESS'
    | 'AUTOMATIC_READY'
    | 'AUTOMATIC_FAILURE'
    | 'UNINSTALL_IN_PROGRESS';
}

export function getEditionStatus(): Promise<EditionStatus> {
  return getJSON('/api/editions/status').catch(throwGlobalError);
}

export function getEditionsList(url: string): Promise<EditionsPerVersion> {
  return corsRequest(url)
    .submit()
    .then(checkStatus)
    .then(parseJSON);
}

export function getLicensePreview(data: {
  license: string;
}): Promise<{
  nextEditionKey: string;
  previewStatus: 'NO_INSTALL' | 'AUTOMATIC_INSTALL' | 'MANUAL_INSTALL';
}> {
  return postJSON('/api/editions/preview', data).catch(throwGlobalError);
}

export function applyLicense(data: { license: string }): Promise<EditionStatus> {
  return postJSON('/api/editions/apply_license', data).catch(throwGlobalError);
}

export function uninstallEdition(): Promise<void | Response> {
  return postJSON('/api/editions/uninstall').catch(throwGlobalError);
}
