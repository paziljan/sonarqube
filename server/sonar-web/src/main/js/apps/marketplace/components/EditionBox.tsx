/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import CheckIcon from '../../../components/icons-components/CheckIcon';
import { Edition, EditionStatus } from '../../../api/marketplace';
import { translate } from '../../../helpers/l10n';

interface Props {
  edition: Edition;
  editionStatus?: EditionStatus;
  onInstall: (edition: Edition) => void;
  onUninstall: () => void;
}

export default class EditionBox extends React.PureComponent<Props> {
  handleInstall = () => this.props.onInstall(this.props.edition);

  render() {
    const { edition, editionStatus } = this.props;
    const installInProgress =
      editionStatus &&
      ['AUTOMATIC_IN_PROGRESS', 'AUTOMATIC_READY'].includes(editionStatus.installationStatus);
    const installReady = editionStatus && editionStatus.installationStatus === 'AUTOMATIC_READY';
    const isInstalled = editionStatus && editionStatus.currentEditionKey === edition.key;
    const isInstalling =
      installInProgress && editionStatus && editionStatus.nextEditionKey === edition.key;
    return (
      <div className="boxed-group boxed-group-inner marketplace-edition">
        {isInstalled &&
        !isInstalling && (
          <span className="marketplace-edition-badge badge badge-normal-size">
            <CheckIcon size={14} className="little-spacer-right text-text-top" />
            {translate('marketplace.installed')}
          </span>
        )}
        {isInstalling && (
          <span className="marketplace-edition-badge badge badge-normal-size">
            {installReady ? translate('marketplace.pending') : translate('marketplace.installing')}
          </span>
        )}
        <div>
          <h3 className="spacer-bottom">{edition.name}</h3>
          <p>{edition.textDescription}</p>
        </div>
        <div className="marketplace-edition-action spacer-top">
          <a href={edition.homeUrl} target="_blank">
            {translate('marketplace.learn_more')}
          </a>
          {!isInstalled && (
            <button disabled={installInProgress} onClick={this.handleInstall}>
              {translate('marketplace.install')}
            </button>
          )}
          {isInstalled && (
            <button
              className="button-red"
              disabled={installInProgress}
              onClick={this.props.onUninstall}>
              {translate('marketplace.uninstall')}
            </button>
          )}
        </div>
      </div>
    );
  }
}
