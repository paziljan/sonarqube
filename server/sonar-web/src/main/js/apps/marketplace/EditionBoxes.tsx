/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { FormattedMessage } from 'react-intl';
import EditionBox from './components/EditionBox';
import LicenseEditionForm from './components/LicenseEditionForm';
import UninstallEditionForm from './components/UninstallEditionForm';
import { Edition, EditionStatus, getEditionsList } from '../../api/marketplace';
import { getEditionsForVersion } from './utils';
import { translate } from '../../helpers/l10n';

export interface Props {
  editionStatus?: EditionStatus;
  editionsUrl: string;
  sonarqubeVersion: string;
  updateCenterActive: boolean;
  updateEditionStatus: (editionStatus: EditionStatus) => void;
}

interface State {
  editions?: Edition[];
  editionsError: boolean;
  loading: boolean;
  installEdition?: Edition;
  openUninstallForm: boolean;
}

export default class EditionBoxes extends React.PureComponent<Props, State> {
  mounted: boolean;
  state: State = { editionsError: false, loading: true, openUninstallForm: false };

  componentDidMount() {
    this.mounted = true;
    this.fetchEditions();
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  fetchEditions = () => {
    this.setState({ loading: true });
    getEditionsList(this.props.editionsUrl).then(
      editionsPerVersion => {
        if (this.mounted) {
          this.setState({
            loading: false,
            editions: getEditionsForVersion(editionsPerVersion, this.props.sonarqubeVersion),
            editionsError: false
          });
        }
      },
      () => {
        if (this.mounted) {
          this.setState({ editionsError: true, loading: false });
        }
      }
    );
  };

  handleOpenLicenseForm = (edition: Edition) => this.setState({ installEdition: edition });
  handleCloseLicenseForm = () => this.setState({ installEdition: undefined });

  handleOpenUninstallForm = () => this.setState({ openUninstallForm: true });
  handleCloseUninstallForm = () => this.setState({ openUninstallForm: false });

  render() {
    const { editionStatus } = this.props;
    const { editions, editionsError, loading, installEdition, openUninstallForm } = this.state;
    if (loading) {
      return <i className="big-spacer-bottom spinner" />;
    }

    if (!editions || editionsError) {
      return (
        <div className="spacer-bottom marketplace-editions">
          <span className="alert alert-info">
            <FormattedMessage
              defaultMessage={translate('marketplace.editions_unavailable')}
              id="marketplace.editions_unavailable"
              values={{
                url: (
                  <a href="https://www.sonarsource.com" target="_blank">
                    SonarSource.com
                  </a>
                )
              }}
            />
          </span>
        </div>
      );
    }

    return (
      <div className="spacer-bottom marketplace-editions">
        {editions.map(edition => (
          <EditionBox
            edition={edition}
            editionStatus={editionStatus}
            key={edition.key}
            onInstall={this.handleOpenLicenseForm}
            onUninstall={this.handleOpenUninstallForm}
          />
        ))}

        {installEdition && (
          <LicenseEditionForm
            edition={installEdition}
            editions={editions}
            onClose={this.handleCloseLicenseForm}
            updateEditionStatus={this.props.updateEditionStatus}
          />
        )}

        {openUninstallForm &&
        editionStatus &&
        editionStatus.currentEditionKey && (
          <UninstallEditionForm
            edition={editions.find(edition => edition.key === editionStatus.currentEditionKey)}
            editionStatus={editionStatus}
            onClose={this.handleCloseUninstallForm}
            updateEditionStatus={this.props.updateEditionStatus}
          />
        )}
      </div>
    );
  }
}
