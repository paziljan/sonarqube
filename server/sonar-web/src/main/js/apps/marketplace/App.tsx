/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import * as PropTypes from 'prop-types';
import { sortBy, uniqBy } from 'lodash';
import Helmet from 'react-helmet';
import Header from './Header';
import EditionsStatusNotif from './components/EditionsStatusNotif';
import EditionBoxes from './EditionBoxes';
import Footer from './Footer';
import PendingActions from './PendingActions';
import PluginsList from './PluginsList';
import Search from './Search';
import {
  getAvailablePlugins,
  getInstalledPluginsWithUpdates,
  getPendingPlugins,
  getPluginUpdates,
  Plugin,
  PluginPending
} from '../../api/plugins';
import { EditionStatus, getEditionStatus } from '../../api/marketplace';
import { RawQuery } from '../../helpers/query';
import { translate } from '../../helpers/l10n';
import { filterPlugins, parseQuery, Query, serializeQuery } from './utils';

export interface Props {
  editionsUrl: string;
  location: { pathname: string; query: RawQuery };
  sonarqubeVersion: string;
  updateCenterActive: boolean;
}

interface State {
  editionStatus?: EditionStatus;
  loading: boolean;
  pending: {
    installing: PluginPending[];
    updating: PluginPending[];
    removing: PluginPending[];
  };
  plugins: Plugin[];
}

export default class App extends React.PureComponent<Props, State> {
  mounted: boolean;

  static contextTypes = {
    router: PropTypes.object.isRequired
  };

  constructor(props: Props) {
    super(props);
    this.state = {
      loading: true,
      pending: {
        installing: [],
        updating: [],
        removing: []
      },
      plugins: []
    };
  }

  componentDidMount() {
    this.mounted = true;
    this.fetchPendingPlugins();
    this.fetchEditionStatus();
    this.fetchQueryPlugins();
  }

  componentDidUpdate(prevProps: Props) {
    if (prevProps.location.query.filter !== this.props.location.query.filter) {
      this.fetchQueryPlugins();
    }
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  fetchQueryPlugins = () => {
    const query = parseQuery(this.props.location.query);
    if (query.filter === 'updates') {
      this.fetchUpdatesOnly();
    } else {
      this.fetchAllPlugins();
    }
  };

  fetchAllPlugins = () => {
    this.setState({ loading: true });
    Promise.all([getInstalledPluginsWithUpdates(), getAvailablePlugins()]).then(
      ([installed, available]) => {
        if (this.mounted) {
          this.setState({
            loading: false,
            plugins: sortBy(uniqBy([...installed, ...available.plugins], 'key'), 'name')
          });
        }
      },
      () => {
        if (this.mounted) {
          this.setState({ loading: false });
        }
      }
    );
  };

  fetchUpdatesOnly = () => {
    this.setState({ loading: true });
    getPluginUpdates().then(
      plugins => {
        if (this.mounted) {
          this.setState({ loading: false, plugins });
        }
      },
      () => {
        if (this.mounted) {
          this.setState({ loading: false });
        }
      }
    );
  };

  fetchPendingPlugins = () =>
    getPendingPlugins().then(
      pending => {
        if (this.mounted) {
          this.setState({ pending });
        }
      },
      () => {}
    );

  fetchEditionStatus = () =>
    getEditionStatus().then(
      editionStatus => {
        if (this.mounted) {
          this.updateEditionStatus(editionStatus);
          setTimeout(this.fetchEditionStatus, 5000);
        }
      },
      () => {}
    );

  updateEditionStatus = (editionStatus: EditionStatus) =>
    this.setState({ editionStatus: editionStatus });

  updateQuery = (newQuery: Partial<Query>) => {
    const query = serializeQuery({ ...parseQuery(this.props.location.query), ...newQuery });
    this.context.router.push({ pathname: this.props.location.pathname, query });
  };

  render() {
    const { editionStatus, loading, plugins, pending } = this.state;
    const query = parseQuery(this.props.location.query);
    const filteredPlugins = query.search ? filterPlugins(plugins, query.search) : plugins;
    return (
      <div className="page page-limited" id="marketplace-page">
        <Helmet title={translate('marketplace.page')} />
        <div className="marketplace-notifs">
          {editionStatus && (
            <EditionsStatusNotif
              editionStatus={editionStatus}
              updateEditionStatus={this.updateEditionStatus}
            />
          )}
          <PendingActions refreshPending={this.fetchPendingPlugins} pending={pending} />
        </div>
        <Header />
        <EditionBoxes
          editionStatus={editionStatus}
          editionsUrl={this.props.editionsUrl}
          sonarqubeVersion={this.props.sonarqubeVersion}
          updateCenterActive={this.props.updateCenterActive}
          updateEditionStatus={this.updateEditionStatus}
        />
        <Search
          query={query}
          updateCenterActive={this.props.updateCenterActive}
          updateQuery={this.updateQuery}
        />
        {loading && <i className="spinner" />}
        {!loading && (
          <PluginsList
            plugins={filteredPlugins}
            pending={pending}
            refreshPending={this.fetchPendingPlugins}
            updateQuery={this.updateQuery}
          />
        )}
        {!loading && <Footer total={filteredPlugins.length} />}
      </div>
    );
  }
}
