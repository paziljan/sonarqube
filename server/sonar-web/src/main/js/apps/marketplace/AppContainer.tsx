/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { connect } from 'react-redux';
import App from './App';
import { getAppState, getGlobalSettingValue } from '../../store/rootReducer';
import './style.css';

const mapStateToProps = (state: any) => ({
  editionsUrl: (getGlobalSettingValue(state, 'sonar.editions.jsonUrl') || {}).value,
  sonarqubeVersion: getAppState(state).version,
  updateCenterActive:
    (getGlobalSettingValue(state, 'sonar.updatecenter.activate') || {}).value === 'true'
});

export default connect(mapStateToProps)(App as any);
