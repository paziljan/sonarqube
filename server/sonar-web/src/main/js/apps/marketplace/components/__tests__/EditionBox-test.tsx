/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { shallow } from 'enzyme';
import { Edition, EditionStatus } from '../../../../api/marketplace';
import EditionBox from '../EditionBox';

const DEFAULT_STATUS: EditionStatus = {
  currentEditionKey: '',
  nextEditionKey: '',
  installationStatus: 'NONE'
};

const DEFAULT_EDITION: Edition = {
  key: 'foo',
  name: 'Foo',
  desc: 'Foo desc',
  download_link: 'download_url',
  more_link: 'more_url',
  request_license_link: 'license_url'
};

it('should display the edition', () => {
  expect(getWrapper()).toMatchSnapshot();
});

it('should display installed badge', () => {
  expect(
    getWrapper({
      editionStatus: {
        currentEditionKey: 'foo',
        nextEditionKey: '',
        installationStatus: 'NONE'
      }
    })
  ).toMatchSnapshot();
});

it('should display installing badge', () => {
  expect(
    getWrapper({
      editionStatus: {
        currentEditionKey: 'foo',
        nextEditionKey: 'foo',
        installationStatus: 'AUTOMATIC_IN_PROGRESS'
      }
    })
  ).toMatchSnapshot();
});

it('should display pending badge', () => {
  expect(
    getWrapper({
      editionStatus: {
        currentEditionKey: '',
        nextEditionKey: 'foo',
        installationStatus: 'AUTOMATIC_READY'
      }
    })
  ).toMatchSnapshot();
});

it('should disable install button', () => {
  expect(
    getWrapper({
      editionStatus: {
        currentEditionKey: '',
        nextEditionKey: 'foo',
        installationStatus: 'AUTOMATIC_IN_PROGRESS'
      }
    })
  ).toMatchSnapshot();
  expect(
    getWrapper({
      editionStatus: {
        currentEditionKey: '',
        nextEditionKey: 'foo',
        installationStatus: 'AUTOMATIC_READY'
      }
    })
  ).toMatchSnapshot();
});

it('should disable uninstall button', () => {
  expect(
    getWrapper({
      editionStatus: {
        currentEditionKey: 'foo',
        nextEditionKey: 'foo',
        installationStatus: 'AUTOMATIC_IN_PROGRESS'
      }
    })
  ).toMatchSnapshot();
});

function getWrapper(props = {}) {
  return shallow(
    <EditionBox
      edition={DEFAULT_EDITION}
      editionStatus={DEFAULT_STATUS}
      onInstall={jest.fn()}
      onUninstall={jest.fn()}
      {...props}
    />
  );
}
