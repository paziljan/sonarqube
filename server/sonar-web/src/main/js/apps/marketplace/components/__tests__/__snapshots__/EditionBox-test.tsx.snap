// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should disable install button 1`] = `
<div
  className="boxed-group boxed-group-inner marketplace-edition"
>
  <span
    className="marketplace-edition-badge badge badge-normal-size"
  >
    marketplace.installing
  </span>
  <div>
    <h3
      className="spacer-bottom"
    >
      Foo
    </h3>
    <p>
      Foo desc
    </p>
  </div>
  <div
    className="marketplace-edition-action spacer-top"
  >
    <a
      href="more_url"
      target="_blank"
    >
      marketplace.learn_more
    </a>
    <button
      disabled={true}
      onClick={[Function]}
    >
      marketplace.install
    </button>
  </div>
</div>
`;

exports[`should disable install button 2`] = `
<div
  className="boxed-group boxed-group-inner marketplace-edition"
>
  <span
    className="marketplace-edition-badge badge badge-normal-size"
  >
    marketplace.pending
  </span>
  <div>
    <h3
      className="spacer-bottom"
    >
      Foo
    </h3>
    <p>
      Foo desc
    </p>
  </div>
  <div
    className="marketplace-edition-action spacer-top"
  >
    <a
      href="more_url"
      target="_blank"
    >
      marketplace.learn_more
    </a>
    <button
      disabled={true}
      onClick={[Function]}
    >
      marketplace.install
    </button>
  </div>
</div>
`;

exports[`should disable uninstall button 1`] = `
<div
  className="boxed-group boxed-group-inner marketplace-edition"
>
  <span
    className="marketplace-edition-badge badge badge-normal-size"
  >
    marketplace.installing
  </span>
  <div>
    <h3
      className="spacer-bottom"
    >
      Foo
    </h3>
    <p>
      Foo desc
    </p>
  </div>
  <div
    className="marketplace-edition-action spacer-top"
  >
    <a
      href="more_url"
      target="_blank"
    >
      marketplace.learn_more
    </a>
    <button
      className="button-red"
      disabled={true}
      onClick={[Function]}
    >
      marketplace.uninstall
    </button>
  </div>
</div>
`;

exports[`should display installed badge 1`] = `
<div
  className="boxed-group boxed-group-inner marketplace-edition"
>
  <span
    className="marketplace-edition-badge badge badge-normal-size"
  >
    <CheckIcon
      className="little-spacer-right text-text-top"
      size={14}
    />
    marketplace.installed
  </span>
  <div>
    <h3
      className="spacer-bottom"
    >
      Foo
    </h3>
    <p>
      Foo desc
    </p>
  </div>
  <div
    className="marketplace-edition-action spacer-top"
  >
    <a
      href="more_url"
      target="_blank"
    >
      marketplace.learn_more
    </a>
    <button
      className="button-red"
      disabled={false}
      onClick={[Function]}
    >
      marketplace.uninstall
    </button>
  </div>
</div>
`;

exports[`should display installing badge 1`] = `
<div
  className="boxed-group boxed-group-inner marketplace-edition"
>
  <span
    className="marketplace-edition-badge badge badge-normal-size"
  >
    marketplace.installing
  </span>
  <div>
    <h3
      className="spacer-bottom"
    >
      Foo
    </h3>
    <p>
      Foo desc
    </p>
  </div>
  <div
    className="marketplace-edition-action spacer-top"
  >
    <a
      href="more_url"
      target="_blank"
    >
      marketplace.learn_more
    </a>
    <button
      className="button-red"
      disabled={true}
      onClick={[Function]}
    >
      marketplace.uninstall
    </button>
  </div>
</div>
`;

exports[`should display pending badge 1`] = `
<div
  className="boxed-group boxed-group-inner marketplace-edition"
>
  <span
    className="marketplace-edition-badge badge badge-normal-size"
  >
    marketplace.pending
  </span>
  <div>
    <h3
      className="spacer-bottom"
    >
      Foo
    </h3>
    <p>
      Foo desc
    </p>
  </div>
  <div
    className="marketplace-edition-action spacer-top"
  >
    <a
      href="more_url"
      target="_blank"
    >
      marketplace.learn_more
    </a>
    <button
      disabled={true}
      onClick={[Function]}
    >
      marketplace.install
    </button>
  </div>
</div>
`;

exports[`should display the edition 1`] = `
<div
  className="boxed-group boxed-group-inner marketplace-edition"
>
  <div>
    <h3
      className="spacer-bottom"
    >
      Foo
    </h3>
    <p>
      Foo desc
    </p>
  </div>
  <div
    className="marketplace-edition-action spacer-top"
  >
    <a
      href="more_url"
      target="_blank"
    >
      marketplace.learn_more
    </a>
    <button
      disabled={false}
      onClick={[Function]}
    >
      marketplace.install
    </button>
  </div>
</div>
`;
